pkg/autoscaler/autoscaler.go

```diff
@@ -199,25 +204,49 @@ func (agg *perPodAggregation) podWeight(now time.Time) float64 {
 // Autoscaler stores current state of an instance of an autoscaler
 type Autoscaler struct {
 	*DynamicConfig
-	key          string
-	target       float64
-	stats        map[statKey]Stat
-	statsMutex   sync.Mutex
-	panicking    bool
-	panicTime    *time.Time
-	maxPanicPods float64
-	reporter     StatsReporter
-	targetMutex  sync.RWMutex
+	key             string
+	namespace       string
+	revisionService string
+	endpointsLister corev1listers.EndpointsLister
+	panicking       bool
+	panicTime       *time.Time
+	maxPanicPods    float64
+	reporter        StatsReporter
+
+	// targetMutex guards the elements in the block below.
+	targetMutex sync.RWMutex
+	target      float64
+
+	// statsMutex guards the elements in the block below.
+	statsMutex sync.Mutex
+	stats      map[statKey]Stat
+
+	// readyPodsMutex guards the elements in the block below.
+	readyPodsMutex sync.RWMutex
+	readyPodsCache map[time.Time]float64
```

Comment:
I'm not quite parsing why we need to use a cache here. The current autoscaler asks for the readyPods exactly once per reporting cycle and thus will never hit this cache?

Reply:
Replied in another comment.